### ISSUE TITLE
CRTM_tutorial unable to compile with CRTM-2.4.0

### ISSUE THREAD

**Issue by kkllww** [StegmannJCSDA/CRTM_tutorial#9]
Hello,
I have compiled crtm-2.4.0 with GNU Fortran (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0 successfully. 
I chose this configuration file `. configuration/gfortran-debug.setup`
Now I am unable to compile CRTM_tutorial with crtm-2.4.0, and have the same problem as the previous user.
This is the makefile of "NakajimaKing":
```
all:
        gfortran -c -I/home/kkllww/crtm/src/Build/crtm_v2.4.0-alpha/include Example_NakajimaKing.f90
        gfortran -L/home/kkllww/crtm/src/Build/crtm_v2.4.0-alpha/lib -lcrtm -o NK.x Example_NakajimaKing.o

clean:
        @rm *.o *.mod

```
Here is the error after running make,  and the error when running make in the Kmatrix_test folder is the same as the error in In function `MAIN__ part.
```
(base) kkllww@pc-PowerEdge-R940xa:~/crtm/CRTM_tutorial/NakajimaKing$ make
gfortran -c -I/home/kkllww/crtm/src/Build/crtm_v2.4.0-alpha/include Example_NakajimaKing.f90
gfortran -L/home/kkllwwn/crtm/src/Build/crtm_v2.4.0-alpha/lib -lcrtm -o NK.x Example_NakajimaKing.o
Example_NakajimaKing.o: In function `signalfile_create.3497':
Example_NakajimaKing.f90:(.text+0x10f): undefined reference to `__file_utility_MOD_get_lun'
Example_NakajimaKing.o: In function `MAIN__':
Example_NakajimaKing.f90:(.text+0x304f): undefined reference to `__crtm_module_MOD_crtm_version'
Example_NakajimaKing.f90:(.text+0x310e): undefined reference to `__message_handler_MOD_program_message'
Example_NakajimaKing.f90:(.text+0x3353): undefined reference to `__crtm_lifecycle_MOD_crtm_init'
Example_NakajimaKing.f90:(.text+0x349f): undefined reference to `__message_handler_MOD_display_message'
Example_NakajimaKing.f90:(.text+0x3512): undefined reference to `__crtm_channelinfo_define_MOD_crtm_channelinfo_subset'
Example_NakajimaKing.f90:(.text+0x3549): undefined reference to `__crtm_channelinfo_define_MOD_crtm_channelinfo_n_channels'
Example_NakajimaKing.f90:(.text+0x3656): undefined reference to `__crtm_channelinfo_define_MOD_crtm_channelinfo_n_channels'
Example_NakajimaKing.f90:(.text+0x36e0): undefined reference to `__crtm_channelinfo_define_MOD_crtm_channelinfo_channels'
Example_NakajimaKing.f90:(.text+0x3ee0): undefined reference to `__message_handler_MOD_display_message'
Example_NakajimaKing.f90:(.text+0x3f56): undefined reference to `__crtm_atmosphere_define_MOD_crtm_atmosphere_create'
Example_NakajimaKing.f90:(.text+0x3f91): undefined reference to `__crtm_atmosphere_define_MOD_crtm_atmosphere_associated'
Example_NakajimaKing.f90:(.text+0x40ea): undefined reference to `__message_handler_MOD_display_message'
Example_NakajimaKing.f90:(.text+0x4163): undefined reference to `__zeeman_input_define_MOD_zeeman_input_setvalue'
Example_NakajimaKing.f90:(.text+0x4210): undefined reference to `__crtm_geometry_define_MOD_crtm_geometry_setvalue'
Example_NakajimaKing.f90:(.text+0x42cb): undefined reference to `__crtm_geometry_define_MOD_crtm_geometry_setvalue'
Example_NakajimaKing.f90:(.text+0x4491): undefined reference to `__crtm_forward_module_MOD_crtm_forward'
Example_NakajimaKing.f90:(.text+0x45d1): undefined reference to `__message_handler_MOD_display_message'
Example_NakajimaKing.f90:(.text+0x47ac): undefined reference to `__crtm_rtsolution_define_MOD_scalar_inspect'
Example_NakajimaKing.f90:(.text+0x4c39): undefined reference to `__crtm_lifecycle_MOD_crtm_destroy'
Example_NakajimaKing.f90:(.text+0x4cf3): undefined reference to `__message_handler_MOD_display_message'
Example_NakajimaKing.f90:(.text+0x4d35): undefined reference to `__crtm_atmosphere_define_MOD_crtm_atmosphere_destroy'
collect2: error: ld returned 1 exit status
makefile:2: recipe for target 'all' failed
make: *** [all] Error 1
```
I checked the *.mod module files of the CRTM according to the previous answer, and the result is as follows, which shows that *.mod files exists.  The libcrtm.a file also exist. The problem of the previous user seems to be because of the version, but I installed the 2.4 version, and the make check command during the build step also showed success. So I don't know where the problem is.
Can anyone give me some advice. 
Thanks for the help!
```
(base) kkllww@pc-PowerEdge-R940xa:~/crtm/src/Build/crtm_v2.4.0-alpha/include$ pwd
/home/kkllww/crtm/src/Build/crtm_v2.4.0-alpha/include
(base) kkllww@pc-PowerEdge-R940xa:~/crtm/src/Build/crtm_v2.4.0-alpha/include$ ls crtm_atmosphere_define.mod
crtm_atmosphere_define.mod
```
```
(base) kkllww@pc-PowerEdge-R940xa:~/crtm/src/Build/crtm_v2.4.0-alpha/lib$ pwd
/home/kkllww/crtm/src/Build/crtm_v2.4.0-alpha/lib
(base) kkllww@pc-PowerEdge-R940xa:~/crtm/src/Build/crtm_v2.4.0-alpha/lib$ ls
libcrtm.a
```

**Comment by StegmannJCSDA** [comment]
Hello @kkllww, thank you for your interest.
This error means that you are unable to link the CRTM with the tutorial example at runtime.
It looks like the example is unable to find the libcrtm.a file. 
One thing you can try is capitalize `-lcrtm` to `-lCRTM` in the makefile and you need to make sure that the paths you are using in the makefile are correct.

**Comment by kkllww** [comment]
Thank you for your prompt reply!
I tried `-lCRTM` and it shows that the file does not exist. The path in the makefile should be correct, my `librtm.a `file is indeed in this folder. I also tried to write the link code in different ways, but the problem still doesn't work out.
Because I followed this link `https://forums.jcsda.org/t/fail-to-make-crtm-test/79/18 `to build the CRTM using ecbuild, so  there is a dynamic link library `libcrtm.so`  in the crtm/build/lib folder, so I changed `gfortran -L /home/kkllww/crtm/src/Build/crtm_v2.4.0-alpha/lib -lcrtm -o NK.x Example_NakajimaKing.o` to `gfortran - o K.x Example_NakajimaKing.o /home/kkllww/crtm/build/lib/libcrtm.so`, and add relevant environment variables and then compile successfully. 
I think it's because I got an error when linking the static library as you said, maybe the environment variable was not set correctly, or the link statement was incorrect. Now I can run the example temporarily with this method and then go back and check the problem.
Thanks again for your help!
